ls1intum/Artemis
# [Feature] Client side validation of Review phase

### Checklist

### Motivation and Context


This PR is addressing [Issue #3844](https://github.com/ls1intum/Artemis/issues/3844):
If a Begin of Student Review is set, but no End of Student Review, no meaningful warning message was displayed to the user so far. This is changed by this PR.
It includes a visual warning below the End of Student Review Box and an additional warning flag next to the disabled submit button.

### Description

I added a method in exam-management.service.ts in order to check that the review phase is defined correctly. Therefore, I introduced a helper attribute in exam.model.ts that flags whether the review phase is defined incorrectly or not.
Besides that, I used this flag to implement the visual warnings described and showed above.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create an Exam
4. Set the Begin of Student Review, but no End of Student Review. A warning next to the End of Student Review Box should appear
5. Scroll down and make sure, that the yellow warning flag is displayed correctly. Additionally, hover over the flag and check whether the appropriate warning is displayed.
6. Also check whether the translation is working appropriately.

### Test Coverage




### Screenshots


Warning beneath End of Student Review:

![image](https://user-images.githubusercontent.com/80622272/128240223-f4224d3b-bfc0-41a5-a891-d273f2ebdfc6.png)

Warning flag next to the safe button:

![image](https://user-images.githubusercontent.com/80622272/128240310-e65b4189-41a7-4898-8a5a-b984f4345b37.png)
